x/oracle/keeper/cycle_list.go

```diff
@@ -61,19 +92,16 @@ func (k Keeper) RotateQueries(ctx context.Context) error {
 			return err
 		}
 		querymeta.CycleList = true
-		querymeta.Expiration = blockTime.Add(querymeta.RegistrySpecTimeframe)
+		querymeta.Expiration = uint64(blockHeight) + querymeta.RegistrySpecBlockWindow
 		return k.Query.Set(ctx, collections.Join(queryId, querymeta.Id), querymeta)
 
 	}
 	// if query has a tip don't generate a new query but extend if revealing time is expired
 	if !querymeta.Amount.IsZero() {
 		querymeta.CycleList = true
-		offset, err := k.GetReportOffsetParam(ctx)
-		if err != nil {
-			return err
-		}
-		if querymeta.Expiration.Add(offset).Before(blockTime) {
-			querymeta.Expiration = blockTime.Add(querymeta.RegistrySpecTimeframe)
+
+		if querymeta.Expiration+offset <= uint64(blockHeight) { // wrong, shouldn't use same query if expired
```

Comment:
this is the case where a query is
- a cycle list query
- has a tip
- is expired
- but didn't get any reveals, otherwise it would have been aggregated and removed in this block
- now that it's the current cycle list query, we want to just keep the tip and re-open this same QueryMeta (we need a better word for QueryMeta)